google/ggrc-core
# GGRC-3742 Improve conflict resolution for simple attributes

# Issue description

The conflict resolution should automatically merge objects where the changes were done to base attributes (e.g. user 1 edits title, user 2 edits description, even though there is a conflict it can still be resolved automatically without input from the user).

The scope of this ticket is limited and could produce false positives (conflict message when there isn't a conflict) when changes were made for complex attributes like custom attributes or control categories. These edge cases will be resolved in separate PRs.

# Steps to test the changes

1. Open object edit modal in two tabs/windows
2. Change object title in one window
3. Change object description in the second
4. Save both modals
Before PR: Conflict message appears when the second object is saved
After PR: Conflict message is not displayed, the objective is saved and the info panes shows the current state (new title AND new description)

# Solution description

This is splitting the work that was done in https://github.com/google/ggrc-core/pull/6407 into more manageable chunks.

On every 409 we check the following for each attribute on the object:
1. Was the attribute changed on the server (we compare it to the current value on the client)
2. Was the attribute changed on the client (we compare the current value to the original state on the client).
3. Was the change made on the client different than the change on the server (if one user changes the description to 'aa' and another user sets the description to 'aa' as well, we don't need to show a warning message)

We show a conflict message only if all 3 points are true.

# Sanity checklist




# PR Review checklist
